src/accelerate/utils/imports.py

```diff
@@ -90,20 +97,19 @@ def is_cuda_available():
 
 
 @lru_cache
-def is_tpu_available(check_device=True):
-    "Checks if `torch_xla` is installed and potentially if a TPU is in the environment"
-    # Due to bugs on the amp series GPUs, we disable torch-xla on them
-    if is_cuda_available():
+def is_torch_xla_available(hardware_types=("TPU", "GPU")):
```

Comment:
This is suddenly a very difficult API to use. Let's just use two boolean values for `tpu` and `gpu` please. These can default to `False` and then we check using `any()` at the end

Reply:
done